src/engine/sparqlExpressions/StringExpressions.cpp

```diff
@@ -619,28 +620,29 @@ struct StrIriDtTag {
 using StrIriTagged = LiteralExpressionImpl<2, StrIriDtTag, IriValueGetter>;
 
 // HASH
-template <auto HashFunc>
-[[maybe_unused]] inline constexpr auto hash =
-    [](std::optional<std::string> input) -> IdOrLiteralOrIri {
-  if (!input.has_value()) {
-    return Id::makeUndefined();
-  } else {
-    std::vector<unsigned char> hashed = HashFunc(input.value());
-    auto hexStr = absl::StrJoin(hashed, "", ad_utility::hexFormatter);
-    return toLiteral(std::move(hexStr));
+template <typename HashFunc>
+struct Hash {
+  IdOrLiteralOrIri operator()(std::optional<std::string> input) const {
+    if (!input.has_value()) {
+      return Id::makeUndefined();
+    } else {
+      std::vector<unsigned char> hashed = HashFunc{}(input.value());
+      auto hexStr = absl::StrJoin(hashed, "", ad_utility::hexFormatter);
+      return toLiteral(std::move(hexStr));
+    }
   }
 };
 
 using MD5Expression =
-    StringExpressionImpl<1, decltype(hash<ad_utility::hashMd5>)>;
+    StringExpressionImpl<1, Hash<decltype(ad_utility::hashMd5)>>;
 using SHA1Expression =
-    StringExpressionImpl<1, decltype(hash<ad_utility::hashSha1>)>;
+    StringExpressionImpl<1, Hash<decltype(ad_utility::hashSha1)>>;
 using SHA256Expression =
-    StringExpressionImpl<1, decltype(hash<ad_utility::hashSha256>)>;
+    StringExpressionImpl<1, Hash<decltype(ad_utility::hashSha256)>>;
 using SHA384Expression =
-    StringExpressionImpl<1, decltype(hash<ad_utility::hashSha384>)>;
+    StringExpressionImpl<1, Hash<decltype(ad_utility::hashSha384)>>;
 using SHA512Expression =
-    StringExpressionImpl<1, decltype(hash<ad_utility::hashSha512>)>;
+    StringExpressionImpl<1, Hash<decltype(ad_utility::hashSha512)>>;
```

Comment:
I think we can get rid of the `decltype` and then also of the inline variables in the header.